src/main/scala/is/hail/expr/ir/Pretty.scala

```diff
@@ -64,7 +64,15 @@ object Pretty {
       sb.append(" " * depth)
       sb += '('
 
-      sb.append(prettyClass(ir))
+      ir match {
+        case read: MatrixRead =>
+          read.reader match {
+            case _: MatrixNativeReader => sb.append("MatrixRead")
```

Comment:
This is confusing -- MatrixRead MatrixRead

Would suggest "Native" or something. And "Range" for MatrixRange

Reply:
I don't quite see what you're suggesting here.

In the deserialized form MatrixRead of Matrix(Native)Reader shows up as just (MatrixRead "path" ...) and MatrixRead of MatrixRangeReader shows up as (MatrixRange 3 7  ...).

I argue this is "right" in the sense that is how the client wants to think about it and the MatrixRead(MatrixReader) business is a implementation detail to simplify some of the analysis code but actually hinders clarity of the interface.

As for the Range suggestion, don't I need to disambiguate Table and Matrix range?

Reply:
ah. I thought that the class name was being written as well -- is that not true?

It would have been simplest to me to write both the name of the node (MatrixRead) and the reader (native). But this is fine too.

Reply:
No, this in lieu of the class name (the default branch).